pkg/udp/conn.go

```diff
@@ -204,12 +204,15 @@ type Conn struct {
 // that is to say it waits on readCh to receive the slice of bytes that the Read operation wants to read onto.
 // The Read operation receives the signal that the data has been written to the slice of bytes through the sizeCh.
 func (c *Conn) readLoop() {
+	ticker := time.NewTicker(c.timeout)
+	defer ticker.Stop()
```

Comment:
I don't understand at all why this would fix anything for now.
If we suppose (even though I don't know why or how) that the problem that somehow ticker.Stop was not called, then I understand why L208 (VS L284) might make a difference.

But then why is L207 needed at all? Why can't L208 just be:

```
    defer c.ticker.Stop()
```

?

Reply:
Also, if we keep on assuming that the problem comes from ticker.Stop not being called, that probably means that c.Close was not always called, which is still a big problem in itself. Because, among other things, that would mean c.listener.conns is not being cleaned up either (which would also be a memleak).

This is all very puzzling.

Reply:
I confirmed c.Close was called (actually multiple times once the timeout was reached), so c.ticker.Stop() was actually being called as well but for some reason the routine and objects were not GCed. I'm puzzled with this as well, my best guess is that the conn, stored in c.listener.conns is not being cleaned up as well but I confirmed it gets removed from the map.

So you are right that we might have more to it than just the ticker, and actually I don't see memory getting back to the starting point level after experiencing a lot of new connections, maybe its conn that is not being GCed for some reason.